File: src/main/java/dev/bluesheep/xeiexporter/ExportUtil.java
package dev.bluesheep.xeiexporter;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import net.minecraft.resources.ResourceLocation;

import java.io.File;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;

public class ExportUtil {
    private static final Gson gson = new GsonBuilder().setPrettyPrinting().create();

    public static void saveExportFile(Object src, Path path) {
        try {
            Files.writeString(path, gson.toJson(src));
        } catch (IOException e) {
            XEIExporter.LOGGER.warn("export failed");
        }
    }

    public static void mkdir(Path path) {
        mkdir(path.toFile());
    }

    public static void mkdir(File path) {
        if (!path.exists()) {
            if (!path.mkdir()) {
                XEIExporter.LOGGER.warn("mkdir failed");
            }
        }
    }

    public static Path resourceLocationToPath(Path base, ResourceLocation recipeId, String extension) {
        var path = base.resolve(recipeId.getNamespace());
        String[] pathArray = (recipeId.getPath() + extension).split("/");
        if (pathArray.length == 1) {
            path = path.resolve(pathArray[0]);
        } else {
            for (int i = 0; i < pathArray.length - 1; i++) {
                path = path.resolve(pathArray[i]);
            }
        }
        path.getParent().toFile().mkdirs();
        return path;
    }

    public static Path resourceLocationToJson(Path base, ResourceLocation resourceLocation) {
        return resourceLocationToPath(base, resourceLocation, ".json");
    }
}

File: src/main/java/dev/bluesheep/xeiexporter/ExportUtil.kt
package dev.bluesheep.xeiexporter

import com.google.gson.Gson
import com.google.gson.GsonBuilder
import net.minecraft.resources.ResourceLocation
import java.io.File
import java.io.IOException
import java.nio.file.Files
import java.nio.file.Path

object ExportUtil {
    private val gson: Gson = GsonBuilder().setPrettyPrinting().create()

    @JvmStatic
    fun saveExportFile(src: Any?, path: Path) {
        try {
            Files.writeString(path, gson.toJson(src))
        } catch (_: IOException) {
            XEIExporter.LOGGER.warn("export failed")
        }
    }

    @JvmStatic
    fun mkdir(path: Path) {
        mkdir(path.toFile())
    }

    @JvmStatic
    fun mkdir(path: File) {
        if (!path.exists()) {
            if (!path.mkdir()) {
                XEIExporter.LOGGER.warn("mkdir failed")
            }
        }
    }

    @JvmStatic
    fun resourceLocationToPath(base: Path, resourceLocation: ResourceLocation, extension: String): Path {
        var path = base.resolve(resourceLocation.namespace)
        val pathArray = "${resourceLocation.path}$extension".split("/")
        if (pathArray.size == 1) {
            path = path.resolve(pathArray[0])
        } else {
            for (i in 0..<pathArray.size - 1) {
                path = path.resolve(pathArray[i])
            }
        }
        path.parent.toFile().mkdirs()
        return path
    }

    @JvmStatic
    fun resourceLocationToJson(base: Path, resourceLocation: ResourceLocation): Path {
        return resourceLocationToPath(base, resourceLocation, ".json")
    }
}
